### DISCUSSION TITLE
use ref inside function with @riverpod

### DISCUSSION THREAD

**Discussion by mafreud** [rrousselGit/riverpod#2462] [Q&A] [answered]
Hi,

Without riverpod generator, I can write as below and use ref.
```dart
final sampleProvider = Provider<String>((ref) {
  return 'hello';
});
``` 

However, with the generator, there are no refs.
```dart
@riverpod
String sample() => 'hello';
```

Any ideas, documentations? Thanks

**Comment by AhmedLSayed9** [comment, marked as answer]
You should pass a ref matching the provider name:
```dart
@riverpod
String sample(SampleRef ref){
  ref.watch(...); 
  return 'hello';
}
```
It'll not compile until you run the generator, but riverpod_lint can help you to easily add it.

**Reply by mafreud** [reply]
Cool!
Thank you always.
